HeartbeatLogging/Sources/Storage.swift

```diff
@@ -91,9 +125,14 @@ final class UserDefaultsStorage: Storage {
     }
   }
 
-  func write(_ value: Data?) throws {
-    if let value = value {
-      defaults.set(value, forKey: key)
+  /// Writes the given data to this object's associated defaults.
+  ///
+  /// When the given `data` is `nil`, this object's associated defaults is emptied.
+  ///
+  /// - Parameter data: The `Data?` to write to this object's associated defaults.
+  func write(_ data: Data?) throws {
+    if let data = data {
+      defaults.set(data, forKey: key)
     } else {
       let emptyData = Data()
       defaults.set(emptyData, forKey: key)
```

Comment:
Hmm, I think you can set `nil` on user defaults now and it will wipe the key instead of having empty data written. Is there a reason why you chose to write data instead of clear it?

Reply:
> Is there a reason why you chose to write data instead of clear it?

The `Storage` protocol is implemented by `UserDefaultsStorage` and `FileStorage`. I was not sure how similar each conforming type should behave. From the conversations in https://github.com/firebase/firebase-ios-sdk/pull/8994, I made `FileStorage`'s `write(nil)` empty the file rather than remove it. When implementing `write(nil)` for `UserDefaultsStorage`, I was not sure if it should follow the similar behavior (empty the value vs removing the key). I chose writing `emptyData` to keep the behavior the same across both types.

I'm curious to know what you think though. I'm unsure if there is a best practice here.

Reply:
FWIW: In terms of how the library uses `Storage`, it does not matter whether `write(nil)` removes the object vs. writing empty data.

Reply:
My mental model for UsersDefaults is a Dictionary in which a key with no value does not makes sense.

My mapping for writing nil to a file system is empty versus a non-existent file which would be created with a remove operation.

I don't feel too strongly though and fine either way, especially since it doesn't seem to make a practical difference for this use case.

Reply:
Thanks team, it was very helpful to hear how y'all interpreted this. I've reworked in c0b7dd969

To summarize the behavior now:

    writing `nil` to:
      - user defaults → removes default
      - file system → empties file